torch/_dynamo/guards.py

```diff
@@ -2851,12 +2851,14 @@ def make_guard_filter_entry(guard):
                     self.guard_manager, output_graph.local_scope
                 )
 
-            # NB for developers: n_iters is chosen to be 50 to achieve
-            # statistical significance.  If you are working on a guard
-            # optimization, it might be a good idea to increase this number for
-            # more stabiilty during development.
+            # NB for developers: n_iters is chosen to be 1 to prevent excessive
+            # increase in compile time. We first do a cache flush to measure the
+            # guard latency more accurately. This cache flush is expensive.
+            # Note  - If you are working on a guard optimization, it might be a
+            # good idea to increase this number for more stabiilty during
+            # development.
             latency = profile_guard_manager(
-                self.guard_manager.root, output_graph.local_scope, 50
+                self.guard_manager.root, output_graph.local_scope, 1
```

Comment:
Do we always profile guards? This seems like it could be expensive?

Reply:
This profiling is during compilation. The maximum I have seen guards take is 3 ms. This is negligible compared to our compile times lol.

Reply:
Looks like the result `latency` is only really used under `TORCH_LOGS="guards"`, should we conditionally run `profile_guard_manager` and give it a higher `n_iters` for more accurate result in the log/tlparse? (I assume you changed 50 to 1 to avoid compilation time overhead with unconditional call to `profile_guard_manager`).

Reply:
With tlparse, we switch on the TORCH_LOGS="guards". Flushing cache is really expensive, so it makes a decent increase in compile time if I run 50 times. Since, tlparse is on by default internally, I dont want the compile time to take a big hit.